### PR TITLE
Warning Banner

### PR DESCRIPTION
**Pull request by IMB11**
Publishing script should work as is, provided by vitepress docs - just need to provide CNAME file via repo settings to display on docs.fabricmc.net

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Sample workflow for building and deploying a VitePress site to GitHub Pages
```
Comment (modmuss50) [inline .github/workflows/publish.yml]: Can I do this please, just a little pedantic about making sure I understand what its doing. And if I make a mistake and it fucks something up its then my fault and not yours 👍

File: .vitepress/theme/components/BannerComponent.vue [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      '--vp-layout-top-height',
+      `${height.value + 16}px`
+    );
+  }
+});
+
+const dismiss = () => {
+  localStorage.setItem(
+    'dismissBanner',
+    (Date.now() + 8.64e7 * 1).toString() // current time + 1 day
```
Comment (modmuss50) [inline .vitepress/theme/components/BannerComponent.vue]: This doesnt seem to be working.

why `* 1`?

`8.64e7` is not clear, just keep it simple and use something like: `24 * 60 * 60 * 1000`

File: .vitepress/theme/components/BannerComponent.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  );
+  document.documentElement.classList.add('banner-dismissed');
+};
+
+onBeforeMount(() => {
+  if (localStorage.getItem('dismissBanner') !== null) {
+    var date = parseInt(localStorage.getItem('dismissBanner')!);
```
Comment (modmuss50) [inline .vitepress/theme/components/BannerComponent.vue]: Nit, store the local stotrage item in a variable, there is a possible (unlikely) race condition here.

File: .vitepress/theme/components/BannerComponent.vue [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    if (Date.now() > date) {
+      dismiss();
```
Comment (modmuss50) [inline .vitepress/theme/components/BannerComponent.vue]: This will set the date forward again, is this expected?

**Comment by IMB11** [comment]
I've completely brain farted on this date logic, one moment.

**Comment by modmuss50** [comment]
It still seems to be faslhing in for a moment, if its not easy to fix its not the end of the world, but it is a little jarring.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**